src/Playwright/Core/Route.cs

```diff
@@ -69,15 +69,37 @@ public Task FulfillAsync(RouteFulfillOptions options = default)
                 options.BodyBytes,
                 options.Path);
 
-            return _channel.FulfillAsync(normalized);
+            return RaceWithPageCloseAsync(_channel.FulfillAsync(normalized));
         }
 
-        public Task AbortAsync(string errorCode = RequestAbortErrorCode.Failed) => _channel.AbortAsync(errorCode);
+        public Task AbortAsync(string errorCode = RequestAbortErrorCode.Failed) => RaceWithPageCloseAsync(_channel.AbortAsync(errorCode));
 
         public Task ContinueAsync(RouteContinueOptions options = default)
         {
             options ??= new RouteContinueOptions();
-            return _channel.ContinueAsync(url: options.Url, method: options.Method, postData: options.PostData, headers: options.Headers);
+            return RaceWithPageCloseAsync(_channel.ContinueAsync(url: options.Url, method: options.Method, postData: options.PostData, headers: options.Headers));
+        }
+
+        internal async Task InternalContinueAsync(RouteContinueOptions options = default)
```

Comment:
Where is this used?

Reply:
BrowserContext.cs onRoute()

Reply:
But not in this PR or? Looks like it is in the other one with a different name:
https://github.com/microsoft/playwright-dotnet/pull/1962/files#diff-2cf3606203604824a9e5ffbc050253e353c746ad7a718e8117c24b149a07508bR354
https://github.com/microsoft/playwright-dotnet/pull/1962/files#diff-4e23d5a0e29b82ff3f03d9d3f1b6c3236e0b336cbefe1598e04d8b1de555ae67R83-R88

Reply:
yeah this was a bad rebase.